### PR TITLE
APPS-1030 Add read write mutex to state file

### PR DESCRIPTION
**Pull request by korotkov-aerospike**

### CODE REVIEW

File: pkg/service/backup_backend_local.go [hunk 1]
```diff
@@ -164,10 +170,11 @@ func toBackupDetails(e fs.DirEntry, prefix string) model.BackupDetails {
 	if err == nil {
 		lastModified = util.Ptr(dirInfo.ModTime())
 	}
+	path := filepath.Join(prefix, e.Name())
 	return model.BackupDetails{
-		Key:          util.Ptr(filepath.Join(prefix, e.Name())),
+		Key:          util.Ptr(path),
 		LastModified: lastModified,
-		Size:         folderSize(prefix),
```
Comment (korotkov-aerospike) [inline pkg/service/backup_backend_local.go:170]: That was the error with file size: we calculated correctly, but for the wrong folder

File: pkg/service/backup_backend_local.go [hunk 2]
```diff
@@ -23,6 +24,7 @@ type BackupBackendLocal struct {
 	stateFilePath        string
 	backupPolicy         *model.BackupPolicy
 	fullBackupInProgress *atomic.Bool // BackupBackend needs to know if full backup is running to filter it out
+	statFileMutex        sync.RWMutex
```
Comment (reugn) [inline pkg/service/backup_backend_local.go:27]: stateFileMutex?

File: pkg/service/backup_backend_local.go [hunk 3]
```diff
@@ -55,6 +57,8 @@ func prepareDirectory(path string) {
 }
 
 func (local *BackupBackendLocal) readState() *model.BackupState {
+	local.statFileMutex.RLock()
+	defer local.statFileMutex.RUnlock()
```
Comment (reugn) [inline pkg/service/backup_backend_local.go:61]: You can unlock right after the `os.ReadFile` operation.

File: pkg/service/backup_backend_s3.go [hunk 4]
```diff
@@ -15,6 +16,7 @@ type BackupBackendS3 struct {
 	stateFilePath        string
 	backupPolicy         *model.BackupPolicy
 	fullBackupInProgress *atomic.Bool // BackupBackend needs to know if full backup is running to filter it out
+	statFileMutex        sync.RWMutex
```
Comment (reugn) [inline pkg/service/backup_backend_s3.go:19]: See the previous naming comment.

**Review by reugn — APPROVED** [review]

**Pull request merged**